app/src/pages/ProtocolDetails/__tests__/ProtocolDetails.test.tsx

```diff
@@ -92,6 +97,7 @@ const render = (path = '/protocols/fakeProtocolId') => {
 
 describe('ODDProtocolDetails', () => {
   beforeEach(() => {
+    vi.mocked(useRunTimeParameters).mockReturnValue([])
```

Comment:
can we use vitest-when for this?

```suggestion
const PROTOCOL_ID = 'mockId'    when(vi.mocked(useRunTimeParameters)).calledWith(PROTOCOL_ID).thenReturn([])
```

Reply:
having issues mocking it this way for some reason, i keep getting an error 🤔 

This error originated in "app/src/pages/ProtocolDetails/__tests__/ProtocolDetails.test.tsx" test file. It doesn't mean the error was thrown inside the file itself, but while it was running.
The latest test that might've caused the error is "renders the parameters screen". It might mean one of the following:
- The error was thrown, while Vitest was running this test.
- If the error occurred after the test had been completed, this was the last documented test before it was thrown.

Reply:
Have you added `vi.resetAllMocks()` to `afterEach()`?

Reply:
it is there 🤔

Reply:
oh! i figured it out. will push the fix